tests/Test.ADAL.NET.Unit/AdalDotNetTests.cs

```diff
@@ -199,7 +199,6 @@ public void Initialize()
                 ResponseMessage = MockHelpers.CreateResiliencyMessage(HttpStatusCode.GatewayTimeout),
             });
 
-            Assert.AreEqual(HttpMessageHandlerFactory.MockHandlersCount(), 2);
```

Comment:
not that I'm against removing spurious asserts, but what's the reason for removing these?

Reply:
I thought you would never ask. :-) Reasons:

1. If I don't remove them, those pre-existing test cases would fail. And that is because this PR will always trigger an Instance Discovery (when the instance cache is initially empty), so one such Instance Discovery mock handler has to be added, thus this pre-existing MockHandlerCount() check becomes off-by-one.

2. I could simply increase that expected value `2` by 1 to compensate that. But that would still break in the future if we introduce yet another cool new feature that adds one more network call. The underlying problem, IMHO, is that our **unit** test cases unnecessarily couple with different features. It just doesn't feel right that when implementing new feature Z, we would be forced to updated previous test cases for feature A which we did not touch for maybe years.

3. In this particular line, that assertion should not be there in the first place. The 2 prior code blocks added 2 mock handlers, and then this line tried to assert the 2 handlers have been added. Wait, wouldn't this be the guaranteed behavior of `AddMockHandler(...)` which, if necessary, would have been tested in `HttpMessageHandlerFactory`'s own unit test cases already? Why would we test it again in this `ExtendedLifetimeExpiredTest()` test case?

For what it's worth, in this PR, I still use similar `Assert.AreEqual(HttpMessageHandlerFactory.MockHandlersCount(), X)` pattern [in some test code](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/796/files#diff-d4bd5ffaf8c9080c752100aae06c1e63R76). But those test cases' scope are minimized to this new Instance Discovery feature only. So the reason no. 2 above would not apply.

Reply:
So for me reason #2, i.e. the "underlying problem" is the big-ticket item here. I'm not suggesting you go after that here, but I *would* suggest filing a work item on the board to address this issue.